openvdb_ax/openvdb_ax/compiler/PointExecutable.cc

```diff
@@ -343,8 +346,11 @@ struct PointExecuterOp
 
         // add attributes based on the order and existence in the attribute registry
         for (const auto& iter : mAttributeRegistry.data()) {
-            const std::string& name = (iter.name() == "P" ? mPositionAttribute : iter.name());
-            addAttributeHandle(args, leaf, name, iter.type(), iter.writes());
+            const std::string* name = mAttributeBindings.dataNameBoundTo(iter.name());
+            // all persistent attributes are in bindings
+            assert(name);
+            if (*name == "P") name = &this->mPositionAttribute;
+            addAttributeHandle(args, leaf, *name, iter.type(), iter.writes());
```

Comment:
Does this need to be done per leaf? Can't we do it once in execute()?

Reply:
This has been updated, made some minor changes to PointExecuterOp and where the P stuff gets handled to only need to use the bindings map in one place( processAttributes())